### PR TITLE
hook up bot adding/removing/editing in team bot view

### PR DESCRIPTION
**Pull request by hyperobject**
This PR adds bot install button to teams bot view and hooks up the edit settings and uninstall buttons in bot row menus.

### CODE REVIEW

File: shared/teams/team/rows/bot-row/bot/container.tsx [hunk 1]
```diff
@@ -16,6 +18,8 @@ type OwnProps = {
 const blankInfo = Constants.initialMemberInfo
 
 type DispatchProps = {
+  _onEdit: (conversationIDKey: Chat2Types.ConversationIDKey, username: string) => void
+  _onRemove: (conversationIDKey: Chat2Types.ConversationIDKey, username: string) => void
```
Comment (songgao) [inline shared/teams/team/rows/bot-row/bot/container.tsx]: I thought explicitly typing `DispatchProps` was not needed any more now that mapStateToProps, mapDispatch, mergeProps are inlined in the `connect` call.

File: shared/teams/team/rows/bot-row/bot/index.tsx [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+      </Kb.Text>
+      <Kb.Text type="BodySmall">
+        &nbsp;• by{' '}
+        {props.ownerTeam ? (
+          <Kb.TeamWithPopup prefix="@" inline={true} teamName={props.ownerTeam} type="BodySmall" />
+        ) : (
+          <Kb.ConnectedUsernames
+            prefix="@"
+            inline={true}
+            usernames={[props.ownerUser ?? props.username]}
```
Comment (songgao) [inline shared/teams/team/rows/bot-row/bot/index.tsx]: Would "by _Unknown_" be better here?

File: shared/teams/team/rows/bot-row/none-row.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  <Box
+    style={{
+      ...globalStyles.flexBoxRow,
+      alignItems: 'center',
+      flexShrink: 0,
+      padding: globalMargins.tiny,
+      width: '100%',
+    }}
+  >
+    <Box style={{...globalStyles.flexBoxRow, alignItems: 'center', flexGrow: 1, justifyContent: 'center'}}>
```
Comment (songgao) [inline shared/teams/team/rows/bot-row/none-row.tsx]: nit: move this to a `const styles`? Same for the box above.

File: shared/teams/team/rows/bot-row/bot/container.tsx (hunk 1 above)
Comment (hyperobject) [reply shared/teams/team/rows/bot-row/bot/container.tsx]: oh good catch, i'll remove that

File: shared/teams/team/rows/bot-row/bot/index.tsx (hunk 2 above)
Comment (hyperobject) [reply shared/teams/team/rows/bot-row/bot/index.tsx]: I think putting the username of the bot is nice so you can get the popup for the bot username (which might contain useful info about the author/source of the bot/whatever)

**Pull request merged**